src/functions.php

```diff
@@ -113,6 +113,120 @@ function stream_for($resource = '', array $options = [])
     throw new \InvalidArgumentException('Invalid resource type: ' . gettype($resource));
 }
 
+
+/**
+ * Return a ServerRequestInterface populated with superglobals :
+ * $_GET
+ * $_POST
+ * $_COOKIE
+ * $_FILES
+ * $_SERVER
+ *
+ * @return ServerRequest
+ */
+function server_request_from_global()
+{
+    $method = !isset($_SERVER['REQUEST_METHOD']) ? 'GET' : $_SERVER['REQUEST_METHOD'];
+    $headers = [];
+    if (function_exists('apache_request_headers')) {
+        $headers = apache_request_headers();
+    }
+    $serverRequest = new ServerRequest(
+        $method,
+        uri_from_global(),
+        $headers,
+        stream_for(fopen('php://input', 'r+')),
+        $_SERVER);
+    $serverRequest = $serverRequest
+        ->withCookieParams($_COOKIE)
+        ->withQueryParams($_GET)
+        ->withParsedBody($_POST)
+        ->withUploadedFiles(uploaded_files_from_global());
+    return $serverRequest;
+}
+
+/**
+ * Return a UriInterface populated with data contained in superglobal $_SERVER
+ *
+ * @return Uri
+ */
+function uri_from_global()
+{
+    $uri = new Uri();
+    $scheme = 'http';
+    if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') {
+        $scheme = 'https';
+    }
+    $host = $_SERVER['SERVER_NAME'];
+    $port = $_SERVER['SERVER_PORT'];
+    $path = $_SERVER['REQUEST_URI'];
+    if (($pos = strpos($path, '?')) !== false) {
+        $path = substr($path, 0, $pos);
+    }
+    $queryString = $_SERVER['QUERY_STRING'];
+    $uri = $uri->withScheme($scheme)
```

Comment:
This seems pretty heavy-- Can you build the URI in one single object instead of over several calls to `with`?